pkg/engine/lifecycletest/update_plan_test.go

```diff
@@ -1714,3 +1714,148 @@ func TestResoucesTargeted(t *testing.T) {
 	}, false, p.BackendClient, nil)
 	assert.Nil(t, res)
 }
+
+// TestReplaceSpecificTargetsPlan checks when replace targets are specified and update targets are not,
+// unspecified targets are not targeted.
+//
+// Tracks a regression where update targets was empty, which is interpreted as an unconstrained update
+// causing all resources to be targeted despite replace targets being specified.
+func TestReplaceSpecificTargetsPlan(t *testing.T) {
+	t.Parallel()
+
+	p := &TestPlan{}
+
+	loaders := []*deploytest.ProviderLoader{
+		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{}, nil
+		}),
+	}
+
+	// Initial state
+	fooVal := "bar"
+
+	createResB := func(monitor *deploytest.ResourceMonitor) {
```

Comment:
We'd generally just do a flag and `if flag { createResource }` rather than changing function references.

Reply:
ack